obspy/io/resp/nrl.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+'''
+nrl.py for navigating, and picking responses from the
+Nominal Response Library.
+
+:copyright:
+    Lloyd Carothers IRIS/PASSCAL, 2016
+:license:
+    GNU Lesser General Public License, Version 3
+    (http://www.gnu.org/copyleft/lesser.html)
+'''
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+# Urllib for py 2 and 3
+from future.standard_library import install_aliases
+install_aliases()  # NOQA
+from urllib.parse import urljoin
+from urllib.request import urlopen
+from os.path import join, dirname
+from ConfigParser import SafeConfigParser
+
+
+class NRL:
+    """
+    Object representing the Nominal Response library.
+    Can be Accessed online from the DMC or with a local copy
+    Decision tree: Usually!!!! depth of tree and order can change!
+    Sensor, make, model, period, sensitivity
+    Datalogger, make, model, gain, sample rate
+
+    The NRL has txt desision files formatted as windows ini
+
+    """
+    URLroot = 'http://ds.iris.edu/NRL/'
+    index = 'index.txt'
+    # Placholder for sample rate and gain
+    SR = object()
+    GAIN = object()
+    dl_shortcuts = {
+        'rt130': ['REF TEK', 'RT 130 & 130-SMA', GAIN, SR],
+        'q330': ['Quanterra', 'Q330SR', GAIN, SR, 'LINEAR AT ALL SPS']}
+    sensor_shortcuts = {
+        'cmg3t': ['Guralp', 'CMG-3T', '120s - 50 Hz', '1500'],
+        'trillium_240_1': [
+            'Nanometrics', 'Trillium 240', '1 - serial numbers < 400'],
+        'l22': ['Sercel/Mark Products', 'L-22D', '5470 Ohms', '20000 Ohms'],
+        'sts2_g3': [
+            'Streckeisen', 'STS-2', '1500', '3 - installed 04/97 to present']
+        }
+
+    def __init__(self, local=True, root=None):
+        if local:
+            # use local copy of NRL on filesystem
+            self.read_ini = self.read_fs_ini
+            self.read_resp = self.read_resp_local
+            self.join = self.join_fs
+            if root is None:
+                raise Exception('Root path of NRL required')
+            else:
+                self.root = root
+        else:
+            # use online, urls etc
+            self.read_ini = self.read_url_ini
+            self.read_resp = self.read_resp_url
+            self.join = urljoin
+            if root is None:
+                self.root = self.URLroot
+            else:
+                self.root = root
+
+    def join_fs(self, path1, path2):
+        return join(dirname(path1), path2)
+
+    def read_ini(self, path):
+        # Replaced with fs or url version on __init__
+        pass
+
+    def read_fs_ini(self, path):
+        # Don't use directly init sets read_ini()
+        cp = SafeConfigParser()
+        with open(path, 'r') as f:
+            cp.readfp(f)
+        return cp
+
+    def read_url_ini(self, url):
```

Comment:
Like stated above, I think it would make sense to have two separate subclasses for the "local" mode and "URL" mode..